File: Domain/Project.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Domain
{
    public class Project
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        public string ProjectName { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime DueDate { get; set; }
        public bool IsActive { get; set; }

        //Relationship
        //public int TeamMemberId { get; set; }
        public ICollection<TeamMember> TeamMember { get; set; }

    }
}

File: client-app/src/app/models/Project.ts
import { ReactNode } from "react";
import { v4 as uuid } from "uuid";

export interface Project {
  id: string;
  projectName: string;
  createdAt: string;
  dueDate: string;
  isActive: string;
}

export interface TeamMember {
  id: { $uuid: any };
  firstName: string;
  lastName: string;
  email: string;
  phoneNumber: string;
  projectName: string;
}


